mypy/semanal.py

```diff
@@ -3680,7 +3683,14 @@ def analyze(self, type: Type) -> None:
             analyzer = TypeAnalyserPass3(self.fail)
             type.accept(analyzer)
 
-    def fail(self, msg: str, ctx: Context, *, blocker: bool = False) -> None:
+    def fail(self, msg: str, ctx: Context, *, blocker: bool = False,
+             implicit_any: bool = False) -> None:
+        if implicit_any:
+            if not self.options.warn_implicit_any or self.errors.file.endswith('.pyi'):
+                return
+            # TempNode, so must have already reported in the first pass
+            if ctx.get_line() == -1:
```

Comment:
This seems like a hack but I couldn't find a way to do it better. Perhaps, someone with more mypy experience may help with that?